File: Arrays/twoSum.js
const twoSum = (nums, target) => {
    let map = {};
    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i];
        if (map[diff] !== undefined) {
            return [map[diff], i]
        } else {
            map[nums[i]] = i;
        }
    }
    return -1;
}

console.log(twoSum([1, 2, 3, 4], 3))

File: Array and Hashing/TwoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val record: MutableMap<Int, Int> = mutableMapOf()
        nums.forEachIndexed { index, value ->
            val remainder = target - value
            
            if (record.containsKey(value)){
                return intArrayOf(record.get(value)!!, index)
            }
            
            record[remainder] = index
        }
        return intArrayOf()
    }
}